sdk/ai/azure-ai-client/samples/inference/sample_trace_chat_completions_client.py

```diff
@@ -0,0 +1,170 @@
+    ]
+
+    response = client.complete(messages=messages, tools=[weather_description, temperature_in_city])
+
+    if response.choices[0].finish_reason == CompletionsFinishReason.TOOL_CALLS:
+        # Append the previous model response to the chat history
+        messages.append(AssistantMessage(tool_calls=response.choices[0].message.tool_calls))
+        # The tool should be of type function call.
+        if response.choices[0].message.tool_calls is not None and len(response.choices[0].message.tool_calls) > 0:
+            for tool_call in response.choices[0].message.tool_calls:
+                if type(tool_call) is ChatCompletionsToolCall:
+                    function_args = json.loads(tool_call.function.arguments.replace("'", '"'))
+                    print(f"Calling function `{tool_call.function.name}` with arguments {function_args}")
+                    callable_func = globals()[tool_call.function.name]
+                    function_response = callable_func(**function_args)
+                    print(f"Function response = {function_response}")
+                    # Provide the tool response to the model, by appending it to the chat history
+                    messages.append(ToolMessage(tool_call_id=tool_call.id, content=function_response))
+                    # With the additional tools information on hand, get another response from the model
+            response = client.complete(messages=messages, tools=[weather_description, temperature_in_city])
+    
+    print(f"Model response = {response.choices[0].message.content}")
+
+
+def main():
+    # Create an Azure AI Client from a connection string, copied from your AI Studio project.
+    # It should have the format "<Endpoint>;<AzureSubscriptionId>;<ResourceGroup>;<WorkspaceName>"
+    ai_client = AzureAIClient.from_connection_string(
+        credential=DefaultAzureCredential(),
+        connection=os.environ["AI_CLIENT_CONNECTION_STRING"],
+    )
+
+    # Or, you can create the Azure AI Client by giving all required parameters directly
+    # ai_client = AzureAIClient(
+    #     credential=DefaultAzureCredential(),
+    #     endpoint=os.environ["AI_CLIENT_ENDPOINT"],
+    #     subscription_id=os.environ["AI_CLIENT_SUBSCRIPTION_ID"],
+    #     resource_group_name=os.environ["AI_CLIENT_RESOURCE_GROUP_NAME"],
+    #     workspace_name=os.environ["AI_CLIENT_WORKSPACE_NAME"],
+    # )
+
+    # Get an authenticated OpenAI client for your default Azure OpenAI connection:
+    client = ai_client.inference.get_azure_openai_client()
+
+    # [START instrument_inferencing]
+    # Speficy a semantic conventions version used for the instrumentation if needed.
+    #instrumentor = ai_client.tracing.create_inference_instrumentor(SemanticConventionsVersion.PREVIEW)
+    
+    # If you do not specify semantic conventions version, then the default version supported by the current
+    # Inference SDK will be returned.
+    instrumentor = ai_client.tracing.create_inference_instrumentor()
```

Comment:
I wonder if this is anti-pattern

ai_client.tracing.instrument_inference()
 
 # do something
 
ai_client.tracing.uninstrument_inference()